common/http-api-client/src/lib.rs

```diff
@@ -90,7 +91,8 @@ impl ClientBuilder {
                 url: url.into_url()?,
                 timeout: None,
                 custom_user_agent: false,
-                reqwest_client_builder: reqwest::ClientBuilder::new(),
+                reqwest_client_builder: reqwest::ClientBuilder::new()
+                    .dns_resolver(Arc::new(HickoryDnsResolver::default())),
```

Comment:
this will not build in `wasm` : (

Reply:
hickory (`reqwest`s choice of alternative resolver supporting DoH & DoT) does not seem to support wasm for now. So this pr only affects non wasm architectures.